### ISSUE TITLE
Cannot update payment methods/sources attached to a subscription

### ISSUE THREAD

**Issue by johnpitchko** [solidusio-contrib/solidus_subscriptions#267]
Might be missing something, but looking through the API, I do not see any capability to update or change the payment method or source attached to a subscription. The [subscription attributes](https://github.com/solidusio-contrib/solidus_subscriptions/blob/abf35035f782d3bbdd3b1a53ebb31d1ba1e9684b/lib/solidus_subscriptions/configuration.rb#L87) do not include any payment values.

**Comment by fthobe** [comment]
Hey @shahmayur001 we should keep a tab on this. It should be tested.

**Comment by fthobe** [comment]
@shahmayur001 is there a lack of implementation on frontend or is not possible at all to change subscription payment methods?

**Comment by mrshahsuvie** [comment]
@fthobe Yes, it is possible to change the subscription payment method, An admin can update the payment method or payment source for a subscription directly from the admin panel based on customer's existing payment methods. 

![image](https://github.com/user-attachments/assets/24b89bd5-8323-4b90-97a8-4de1b8c75be3)


**issue found**: The payment source and payment methods are updated but the events of those changes are not getting created.

![image](https://github.com/user-attachments/assets/27471834-697f-432e-9353-0055b4671c46)

**Comment by shahmayur001** [comment]
@fthobe Yes, it is possible to change the subscription payment method, An admin can update the payment method or payment source for a subscription directly from the admin panel based on customer's existing payment methods. 

![image](https://github.com/user-attachments/assets/24b89bd5-8323-4b90-97a8-4de1b8c75be3)


**issue found**: The payment source and payment methods are updated but the events of those changes are not getting created.

![image](https://github.com/user-attachments/assets/27471834-697f-432e-9353-0055b4671c46)

**Comment by shahmayur001** [comment]
@fthobe 

1) Changing the payment method from the customer's existing payment methods is working, but it does not create an event for that action. Like events are generated when a subscription is created or other modifications are made.

2) It is true that users currently have no way to update their subscription's payment method from the frontend.
